### PR TITLE
Add new_shopify_cli_app flag for bypassing prompts when using shopify…

### PR DESCRIPTION
**Pull request by rezaansyed**
Related to this: https://github.com/Shopify/shopify-app-cli/pull/1086#discussion_r581119425

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `docs/`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

**Comment by NabeelAhsen** [comment]
This flag should likely be available to the install_generator as well, because it's responsible for calling `shop_model` generator.

**Review by paulomarg — APPROVED** [review]
LGTM, just checking: is the standard here to use snake_case for the parameter name? The Polaris guidelines point towards preferring kebab-case for command line arguments.

**Comment by NabeelAhsen** [comment]
> LGTM, just checking: is the standard here to use snake_case for the parameter name? The Polaris guidelines point towards preferring kebab-case for command line arguments.

Hey Paulo, I'll be taking over this PR. Thank you for point that out! I'll make this kebab-case + there are a few more changes needed + tests written before this should ship.

**Comment by NabeelAhsen** [comment]
> LGTM, just checking: is the standard here to use snake_case for the parameter name? The Polaris guidelines point towards preferring kebab-case for command line arguments.

So it turns out that the options can be passed in on the command line using kebab-case! Rubocop complains if any variable isn't snake-case, so there appear to be some magic that rewrites these options into underscored symbols.

So something like this works for me locally:

```terminal
$ rails generate shopify_app --new-shopify-cli-app
```

File: lib/generators/shopify_app/shopify_app_generator.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def initialize(args, *options)
 
       def run_all_generators
         generate("shopify_app:install #{@opts.join(' ')}")
-        generate("shopify_app:shop_model")
+        generate("shopify_app:shop_model #{@pts.join(' ')}")
```
Comment (rezaansyed) [inline lib/generators/shopify_app/shopify_app_generator.rb:12]: should be `@opts`

File: lib/generators/shopify_app/shop_model/shop_model_generator.rb [hunk 2]
```diff
@@ -49,8 +49,8 @@ def create_shop_fixtures
 
       private
 
-      def force_access_scopes_migration?
-        options['force_access_scopes_migration']
+      def new_shopify_cli_app?
+        options['new_shopify_cli_app']
```
Comment (paulomarg) [inline lib/generators/shopify_app/shop_model/shop_model_generator.rb:53]: Interesting, I didn't know Rails handled converting the param format.

**Review by paulomarg — APPROVED** [review]

**Pull request merged**